lib/config/flat-config-schema.js

```diff
@@ -143,6 +137,23 @@ function normalizeRuleOptions(ruleOptions) {
     return structuredClone(finalOptions);
 }
 
+/**
+ * Determines if an object has any methods.
+ * @param {Object} object The object to check.
+ * @returns {boolean} `true` if the object has any methods.
+ */
+function hasMethod(object) {
+
+    for (const key of Object.keys(object)) {
+
+        if (typeof object[key] === "function") {
+            return true;
+        }
+    }
+
+    return false;
+}
```

Comment:
Perhaps we should also update https://eslint.org/docs/latest/extend/custom-parsers#methods-in-custom-parsers section with a requirement that `parse`/`parseForESLint` must be own properties of the parser object, since this checks only own properties.